recipe-server/client/control/tests/reducers/test_controlAppReducer.js

```diff
@@ -207,18 +211,53 @@ describe('controlApp reducer', () => {
       ...initialState,
       recipes: {
         ...initialState.recipes,
-        list: [{
-          id: 1,
-          name: 'Lorem Ipsum',
+        list: {
+          1: {
+            id: 1,
+            name: 'Lorem Ipsum',
+            enabled: true,
+          },
+          2: {
+            id: 2,
+            name: 'Dolor set amet',
+            enabled: true,
+          },
+        },
+      },
+    });
+  });
+
+  describe('Recipe Creation and Updating', () => {
+    it('only has one recipe in the `list` after creating and updating a new recipe', async () => {
```

Comment:
This test looks good, and is probably a valuable thing to have, but I don't think it actually tests the ghost recipe problem. I copied this test and added it to the master branch without any of the other changes. After modifying the call to expect to be the below the test passed. (I changed it to expect the old, array based format, instead of the objects).

```
expect(store.recipes.list).toEqual([
  {
    id: 3,
    name: 'Updated recipe name',
    enabled: true,
  },
]);
```

Since this test didn't fail before this patch, I think we'll have to go with the more complex UI test if we want to verify this.